### ISSUE TITLE
kr1: denas aura's range increase doesnt apply to some towers

### ISSUE THREAD

**Issue by sunnydaler** [cigumo/krli#1497]
Tesla https://cdn.discordapp.com/attachments/567811170342535188/807665377467170846/aura_on_tesla.mp4

Bertha https://cdn.discordapp.com/attachments/567811170342535188/807665378053587004/aura_on_bertha.mp4

**Comment by Kakaaakak** [comment]
The issue with King Denas's Royal Edict lies within how attacks.range is used in artillery's script.
Royal Edict applies mod_denas_tower to the affected towers. This mod increases tt.attacks.range by multiplying it with modifier's range_factor value.
From scripts.mod_denas_tower.insert:

> if this.range_factor then
> 		target.attacks.range = target.attacks.range * this.range_factor
> end

Most towers use tt.attacks.range for basic attack range, while Big Bertha, for example, ignores that value and only uses the value listed in tt.attacks.list[1].range. Same goes for Tesla.
As Big Bertha's and Tesla's attacks do not care for tt.attacks.range, but only check the tt.attacks.list[1].range, the modifier does nothing for them. It does, however change their attack speed, because in modifier's script it actually specifies that it affects target.attacks.list[1].cooldown.
I fixed the issue by changing modifier's range-increasing part (the one written above) to this:

> if this.range_factor then
> 		if target.attacks.list[1].range then
> 		target.attacks.list[1].range = target.attacks.list[1].range * this.range_factor
> 		end
> 		target.attacks.range = target.attacks.range * this.range_factor
> end

Basically, I added a second check to see if tower has target.attacks.list[1].range specified, and changing that value by using the same formula. That fixes the issue with basic attacks, though I'm not sure if it does anything for tower's abilities.

**Comment by Mmlh1** [comment]
Looks like Tesla could just use its basic attack range, since it does not have multiple attacks it loops over. For Bertha, this is a bit more difficult, and this solution would work for the basic attack.
